lib/web_ui/lib/src/engine/html_image_codec.dart

```diff
@@ -180,6 +180,9 @@ class HtmlImage implements ui.Image {
       ctx.drawImage(imgElement, 0, 0);
       final html.ImageData imageData = ctx.getImageData(0, 0, width, height);
       return Future<ByteData?>.value(imageData.data.buffer.asByteData());
+    } else if (format == ui.ImageByteFormat.rawStraightRgba) {
+      throw UnsupportedError(
+        'Image.toByteData(format: ui.ImageByteFormat.rawStraightRgba) not yet implemented for HTML');
```

Comment:
I have filed an issue (https://github.com/flutter/flutter/issues/89094) for it and add a `TODO` here. I will try to implement it next time.

Reply:
I tried to use webGL to get straight RGBA and it worked. But when writing test cases, I accidentally discovered that the `rawRgba` format in Html returns straight RGBA instead of premultiplied RGBA. So I filed a PR https://github.com/flutter/engine/pull/28357 and an Issue https://github.com/flutter/flutter/issues/89128